doc/development/adding_templates.rst

```diff
@@ -0,0 +1,102 @@
+How to add a new template
+-------------------------
+
+The following steps will help you to add your favourite circuit ansatz to
+PennyLane's :mod:`template <pennylane.template>` library.
+
+PennyLane internally distinguishes :doc:`different types </introduction/templates>` of templates, such as
+:mod:`embeddings <pennylane.templates.embeddings>`, :mod:`layers <pennylane.templates.layers>`,
+:mod:`state_preparations <pennylane.templates.state_preparations>` or
+:mod:`subroutines <pennylane.templates.subroutines>`.
+Here we will use the template ``MyNewTemplate`` as an example, and you need to replace ``<templ_type>`` with the
+correct template type.
+
+1. **Add the template** by adding a new file ``my_new_template.py`` to the correct ``templates/<templ_type>/``
+   subdirectory. The file contains your new template, which is a function that defines a sequence of
+   quantum gates (without measurements).
+
+   For example, this is a very basic template applying an :class:`~.RX`
+   gate to each wire:
+
+   .. code-block:: python
+
+        import pennylane as qml
+        from pennylane.templates import template  # import the decorator
+
+        @template
+        def MyNewTemplate(weights, wires):
+
+            # Check that the inputs have the correct format
+            # ...
+
+            for wire, weight in zip(wires, weights):
+                qml.RX(weight, wires=wire)
+
+   Since the template is called within a :ref:`quantum function <intro_vcirc_qfunc>`,
+   it can only contain information processing that is allowed
+   inside a quantum functions.
+
+   Make sure you consider the following:
+
+   * *Choose the name carefully.* Good names tell the user what a template is good for,
+      or what architecture it implements. To be consistent with other quantum operations
+      (which are classes), the function name (i.e., ``MyNewTemplate``) is written in camel case. 
+
+   * *Register the template.* A template is "registered" by using the :mod:`template <pennylane.template>`
```

Comment:
<pennylane.template**s**> ??

Reply:
I was wondering for a second too, but this is not referring to the ``templates`` module, but to the function ``template``. Good observation though.